services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -126,6 +157,15 @@ public static JDBCClient dataSource(final Vertx vertx, final JdbcProperties data
         if (dataSourceProperties.getMaximumPoolSize() != null) {
             config.put("max_pool_size", dataSourceProperties.getMaximumPoolSize());
         }
+        if (dataSourceProperties.getMinimumPoolSize() != null) {
+            config.put("min_pool_size", dataSourceProperties.getMinimumPoolSize());
+        }
+        if (dataSourceProperties.getInitialPoolSize() != null) {
```

Comment:
should we also add a check whether the initial pool size is indeed between min and max?

Reply:
I thought about this as well, but decided against it because, since every value could be null (via default constructor or setter methods), it would be very complicated to cover every edge case meaningfully.

Reply:
when we change the types of these properties to `int`, there shouldn't be an issue with checking the boundaries anymore, right?

Reply:
yes, but then we would have to get rid of the default constructor or add the default values there as well.

Reply:
Yes, that is what we do in other properties classes as well, e.g. `ServiceConfigProperties`. This is not pretty but at least it would be consistent. The overall issue is that we had introduced the Options/Properties duality during the time of the migration from Spring to Quarkus. Spring heavily uses properties, Quarkus uses the interface based Options approach. We will need to start tidying that up ...

Reply:
ok, I was already wondering why that is. I added the defaults to the Properties class as well, changed the types from Integer to int and added a value validation.